src/wfinit.c

```diff
@@ -1312,6 +1312,14 @@ JAPANEND
    win.rc.right -= win.rc.left;
    win.rc.bottom -= win.rc.top;
 
+   // We need to know about all reaprse tags
+   hNtdll = LoadLibrary(L"ntdll.dll");
+   if (hNtdll)
+   {
+      pfnRtlSetProcessPlaceholderCompatibilityMode = (RtlSetProcessPlaceholderCompatibilityMode_t)GetProcAddress(hNtdll, "RtlSetProcessPlaceholderCompatibilityMode");
+      if (pfnRtlSetProcessPlaceholderCompatibilityMode)
+         pfnRtlSetProcessPlaceholderCompatibilityMode(PHCM_EXPOSE_PLACEHOLDERS);
+   }

```

Comment:
How is the behavior of PHCM_APPLICATION_DEFAULT determined?  Wouldn't it be better to change that?

Reply:
(Sorry for being MIA, I've been busy with day job)

I had the same reaction and looked this up the other day.  PHCM_APPLICATION_DEFAULT is zero, which is used because the PEB is initialized with RtlZeroMemory.  The meaning is provided by `HsmOsGetPlaceholderCompatMode`, which is a clean function that's easy to read.  But the high level point is that, if it's not a sync process or an OS process, it's going to get hidden placeholder behavior unless it explicitly asks to observe placeholders.

Reply:
Correct. PHCM_APPLICATION_DEFAULT would not give info on reparse points for OneDrive. 
We need this info, so that we can follow reparse points during goto-Cache fill-up